src/control/agent/main.go

```diff
@@ -24,41 +24,92 @@
 package main
 
 import (
-	"flag"
+	"github.com/daos-stack/daos/src/control/client"
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/drpc"
+	"github.com/daos-stack/daos/src/control/log"
+	"github.com/jessevdk/go-flags"
 	"os"
 	"os/signal"
 	"path/filepath"
 	"syscall"
-
-	"github.com/daos-stack/daos/src/control/common"
-	"github.com/daos-stack/daos/src/control/drpc"
-	"github.com/daos-stack/daos/src/control/log"
 )
 
-var (
-	runtimeDir = flag.String("runtime_dir", "/var/run/daos_agent", "The path to runtime socket directory for daos_agent")
-	logFile    = flag.String("logfile", "/tmp/daos_agent.log", "Path for the daos agent log file")
+const (
+	agentSockName        = "agent.sock"
+	daosAgentDrpcSockEnv = "DAOS_AGENT_DRPC_DIR"
 )
 
+type cliOptions struct {
+	ConfigPath string `short:"o" long:"config-path" description:"Path to agent configuration file"`
+	RuntimeDir string `short:"s" long:"runtime_dir" description:"Path to agent communications socket"`
+	LogFile    string `short:"l" long:"logfile" description:"Full path and filename for daos agent log file"`
+}
+
+var opts = new(cliOptions)
+
 func main() {
 	if agentMain() != nil {
 		os.Exit(1)
 	}
 }
 
+// applyCmdLineOverrides will overwrite Configuration values with any non empty
+// data provided, usually from the commandline.
+func applyCmdLineOverrides(c *client.Configuration, opts *cliOptions) {
```

Comment:
You have two different implementations of this function. Any reason why the functionality of applyCmdLineOverrides in agent/main.go  and applyCmdLineOverrides in dmg/main.go can't be combined and placed in client/config.go?

Reply:
An earlier implementation had the applyCmdLineOverrides functionality in two separate functions within client/config.go.  I was asked to remove that place it into the agent and dmg.  The cliOptions definitions are different between the agent and dmg, which makes combining the functionality into a single function somewhat cumbersome.

Reply:
Gotcha. Thanks for the context on that.